### PR TITLE
Allow customization of aud claim with JWT Authentication

### LINKED ISSUE #21445: Allow customization of `aud` claim when using `private_key_jwt` with OIDC IdP
### Description

Currently the token generated always sets the `aud` claim to the token url.

The request is to allow the `aud` claim to be set to a configurable value.

https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java#L423-L434

### Discussion

_No response_

### Motivation

The OIDC IdP I would like to integrate with expects the `aud` claim to be a different value from the token url. 

It actually expects it to be the `issuer` value in their `openid-configuration`. Allowing it to be customizable will allow integration with such IdPs.

### Details

_No response_

### PR DESCRIPTION
**Pull request by justin-tay**
Closes #21445

This allows integration with IdPs that require the audience claim on the JWT used for JWT Authentication to be a value other than the token endpoint for example the value might be the value of the issuer.

### CODE REVIEW

**Comment by pedroigor** [comment]
FTR, the specs also state that using the token endpoint URL is optional and other values (e.g.: issuer) can be used too. See https://datatracker.ietf.org/doc/html/rfc7523#section-3.

File: js/apps/admin-ui/cypress/support/pages/admin-ui/manage/identity_providers/ProviderBaseAdvancedSettingsPage.ts [hunk 1]
```diff
@@ -29,10 +29,10 @@ export enum PromptSelect {
 }
 
 export enum ClientAuthentication {
-  post = "Client secret sent as basic auth",
-  basicAuth = "Client secret as jwt",
-  jwt = "JWT signed with private key",
-  jwtPrivKey = "Client secret sent as post",
+  post = "Client secret sent as post",
+  basicAuth = "Client secret sent as basic auth",
+  jwt = "Client secret as jwt",
```
Comment (pedroigor) [inline js/apps/admin-ui/cypress/support/pages/admin-ui/manage/identity_providers/ProviderBaseAdvancedSettingsPage.ts:34]:
```suggestion
  jwt = "JWT signed with client secret",
```

File: services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java [hunk 2]
```diff
@@ -427,7 +427,11 @@ protected JsonWebToken generateToken() {
         jwt.type(OAuth2Constants.JWT);
         jwt.issuer(getConfig().getClientId());
         jwt.subject(getConfig().getClientId());
-        jwt.audience(getConfig().getTokenUrl());
+        String audience = getConfig().getClientAssertionAudience();
+        if (audience == null || audience.isEmpty()) {
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java:431]: Better use `org.keycloak.utils.StringUtil#isNotBlank` to make sure the string is also trimmed before comparison.

**Review by pedroigor — COMMENTED** [review]
@justin-tay Thanks. LGTM.

@keycloak/ui-maintainers Could you please review the UI changes? 

In general, changes to messages look good to me. Not sure if test failure is related?

**Review by pedroigor — COMMENTED** [review]
@justin-tay Thanks. I'm good with the changes.

Let's see what @keycloak/ui-maintainers have to say.

**Review by jonkoops — DISMISSED** [review]
Changes to the front-end look good to me.

**Comment by jonkoops** [comment]
@justin-tay please do not update PRs with rebases unless needed, as this might reset the review status and create additional notifications for us.

**Review by jonkoops — APPROVED** [review]
Changes on the front-end still look good to me.

**Review by pedroigor — APPROVED** [review]

**Pull request merged**